discovery/statichosts/lib.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package statichosts
+
+// StaticHostList is a static list of hosts to bootstrap from.
+type StaticHostList struct {
+	hosts []string
+}
+
+// Hosts just returns the static list of hosts that was provided to this struct
+// on construction.
+func (p *StaticHostList) Hosts() ([]string, error) {
+	return p.hosts, nil
+}
+
+// New creates a provider with a static list of hosts.
+func New(hosts []string) *StaticHostList {
```

Comment:
Have you considered using [`hosts ...string`](https://golang.org/ref/spec#Passing_arguments_to_..._parameters) here? It makes it easier to use if you hardcode the hosts in your application since you don't have to initialize the slice your self.

If you happen to already have a slice you can still pass it in to this function via `New(yourHostsLists...)`